pkg/query/remote_engine.go

```diff
@@ -289,8 +306,15 @@ func (r *remoteQuery) Exec(ctx context.Context) *promql.Result {
 				warnings.Add(errors.New(warn))
 				continue
 			}
+			if s := msg.GetStats(); s != nil {
+				qryStats = *s
+				continue
+			}
 
 			ts := msg.GetTimeseries()
```

Comment:
since old root queriers dont have the ==nil check ~ this will probably break if leaf queriers are updated to send stats first. One should update root queriers first and then leaf queriers probably.

Reply:
We can add this as a note to the CHANGELOG for the distributed engine use case. I don't think many people use this mode yet, but it is good to point out the rollout strategy.

Reply:
Yeah! I dont think its a big issue ~ just wanted to point it out; We can add a small warning in release notes later